app/Http/Controllers/Service/Admin/CentreUsersController.php

```diff
@@ -223,7 +223,10 @@ private function syncCentres($request, CentreUser $cu)
 
     public function download()
     {
-        $workers = CentreUser::get()->sortBy('homeCentre.name');
+        $workers = CentreUser::get()->sortBy(function($centreUser) {
+            return sprintf('%-12s%s', $centreUser->homeCentre->name, $centreUser->name);
```

Comment:
ah! hashing the  two together and sorting them
I read that as 
parameter 1, left jsutified, first 12 characters,
Parameter 2, not justified. 

why the twelve characters, and why the left justify?

Reply:
Also,
could you return 
"string1#string2"
using the delimeter "#"?

Reply:
I don't really get it what's going on here. But here goes an explanation I gleaned from teh interwebs. If you need the sortBy against multiple columns, you probably need to space pad them to ensure that "ABC" and "DEF" comes after "AB" and "DEF", hence the sprint right padded for each column up to the column's length (at least for all but the last column)

Reply:
> Also,
> could you return
> "string1#string2"
> using the delimeter "#"?

this doesn't seem to do anything. my version, below 👇 

`$workers = CentreUser::get()->sortBy(function() {
            return 'homeCentre.name#name';
        });`

Reply:
alas, found an easier solution. had to read an article I skipped the first time around...

Reply:
it's counterintuitive but if you have users collection and you want to sort by last name, first name, you'd think you can do `dd($users->sortBy('last_name')->sortBy('first_name'));`

actually, you need to put them in reverse order
`dd($users->sortBy('first_name')->sortBy('last_name'));`

👀 🕵️‍♂️

Reply:
something like
```
$workers = CentreUser::get()->sortBy(function ($centreUser) {
  return $centreUser->homeCentre->name . "#" . $centreUser-name
}
```